torch/nn/modules/activation.py

```diff
@@ -768,11 +774,6 @@ def forward(self, query, key, value, key_padding_mask=None,
                 q_proj_weight=self.q_proj_weight, k_proj_weight=self.k_proj_weight,
                 v_proj_weight=self.v_proj_weight)
         else:
-            if not hasattr(self, '_qkv_same_embed_dim'):
```

Comment:
the PR description of "user has to do this" cannot be known by the user magically.

When user does `model = torch.load('model.pt')`, they have no idea that they are supposed to create a new MultiHeadAttention and load the state dict into it.
This is not a good fix, just to make the JIT happy.

Reply:
With the current code, users don't need to worry the BC breaking. We will communicate with the JIT team.